contracts/casino/src/casino.cpp

```diff
@@ -99,53 +107,65 @@ uint64_t casino::get_game_id(name game_account) {
 
 void casino::verify_game(uint64_t game_id) {
     check(platform::read::is_active_game(get_platform(), game_id), "the game was not verified by the platform");
-    check(is_active_game(game_id), "the game is not run by the casino");
-}
-
-void casino::verify_account(name game_account) {
-    verify_game(get_game_id(game_account));
+    check(is_active_game(game_id), "the game is paused");
 }
 
 void casino::withdraw(name beneficiary_account, asset quantity) {
     require_auth(get_owner());
     const auto ct = current_time_point();
     const auto account_balance = token::get_balance(_self, core_symbol);
+    // in case game developers screwed it up
+    const auto game_profits_sum = std::max(zero_asset, gstate.game_profits_sum);
 
-    if (account_balance > gstate.game_active_sessions_sum + gstate.game_profits_sum) {
+    if (account_balance > gstate.game_active_sessions_sum + game_profits_sum) {
         const asset max_transfer = account_balance - gstate.game_active_sessions_sum - gstate.game_profits_sum;
```

Comment:
Shouldn't `gstate.game_profits_sum` here be replaced with `game_profits_sum`?

Reply:
fixed